src/Application/Commands/ArchiveShoppingList/ArchiveShoppingListCommand.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lindyhopchris\ShoppingList\Application\Commands\ArchiveShoppingList;
+
+use Lindyhopchris\ShoppingList\Persistance\ShoppingListRepositoryInterface;
+
+class ArchiveShoppingListCommand implements ArchiveShoppingListInterface
+{
+    /**
+     * @var ShoppingListRepositoryInterface
+     */
+    private ShoppingListRepositoryInterface $repository;
+
+    public function __construct(ShoppingListRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(ArchiveShoppingListModel $model): void
+    {
+        $list = $this->repository->findOrFail(
+            $model->getList(),
+        );
+
+        $list->isArchived(); //This property needs to be added to ShoppingList.
```

Comment:
Comment: it's more that here you are going to set the archived status, `isArchived` is the "getter", not the "setter". Just mentioning that now; it's not strictly necessary to change this now because you'll actually get this working in a subsequent micro-step (think 3 as far as I remember).